src/LuaTransformer.ts

```diff
@@ -477,6 +477,19 @@ export class LuaTransformer {
             }
         }
 
+        // You cannot extend LuaTable classes
+        if (extendsType) {
+            const decorators = tsHelper.getCustomDecorators(extendsType, this.checker);
+            if (decorators.has(DecoratorKind.LuaTable)) {
+                throw TSTLErrors.InvalidExtendsLuaTable(statement);
+            }
+        }
+
+        // LuaTable classes must be declared
+        if (decorators.has(DecoratorKind.LuaTable) && !tsHelper.isDeclared(statement)) {
```

Comment:
Class may considered declared because of it's context
```suggestion
        if (decorators.has(DecoratorKind.LuaTable) && (ts.getCombinedModifierFlags(statement) & ts.ModifierFlags.Ambient) === 0) {
```

Reply:
Not sure how to test this but good pickup